### PR TITLE
Check for invalid PIN

### LINKED ISSUE #904: Device with 1.9.0 FW gets wiped when you enter PIN code with 0
**Describe the bug**
Device with 1.9.0 FW gets wiped when you enter PIN code with 0

**Firmware version or revision**
1.9.0

**Desktop/smartphone setup (please complete the following information):**
trezorctl 12.0

**To Reproduce**
1. create new wallet
2. set PIN 
3. trezorctl set wipe-code
4. enter 90 when asked for PIN code
5. device gets wiped

**Expected behavior**
device don't get wiped

### PR DESCRIPTION
**Pull request by andrewkozlik**
Fixes #904.

### CODE REVIEW

File: python/src/trezorlib/ui.py [hunk 1]
```diff
@@ -95,8 +95,10 @@ def get_pin(self, code=None):
                 pin = prompt("Please enter {}".format(desc), hide_input=True)
             except click.Abort:
                 raise Cancelled from None
-            if not pin.isdigit():
-                echo("Non-numerical value provided, please try again")
+            if any(d not in "123456789" for d in pin):
```
Comment (matejcik) [inline python/src/trezorlib/ui.py]: please change this check also in `client.py` `_callback_pin`

Comment (andrewkozlik) [reply python/src/trezorlib/ui.py]: Done https://github.com/trezor/trezor-firmware/pull/906/commits/4d46fd078e50f01ec53c7bed2ced7caa472dc4b7

**Comment by andrewkozlik** [comment]
@prusnak just glance over the legacy changes.

**Comment by matejcik** [comment]
shouldn't there also be tests for empty / too long pin?
or do those exist already?

other than that LGTM

File: tests/device_tests/test_msg_change_wipe_code_t1.py [hunk 2]
```diff
@@ -174,3 +175,25 @@ def test_set_pin_to_wipe_code(client):
     assert client.features.pin_protection is False
     resp = client.call_raw(messages.GetAddress())
     assert isinstance(resp, messages.Address)
+
+
+def test_set_wipe_code_invalid(client):
```
Comment (matejcik) [inline tests/device_tests/test_msg_change_wipe_code_t1.py]: perhaps `@pytest.mark.parametrize("invalid_pin", ("1204", "", "1234567891"))` ? dtto for other test cases

Comment (andrewkozlik) [reply tests/device_tests/test_msg_change_wipe_code_t1.py]: Good idea. Done.
